### ISSUE TITLE
can this node-gyp warning be dealt with?

### ISSUE THREAD

**Issue by eric-g-97477** [nodejs/nan#962]
I am using node 18.18.2 and yarn 3.2.4.

If I do:

1. mkdir test
2. cd test
3. yarn init
4. yarn add nan

I get the warning:

```
➤ YN0032: │ nan@npm:2.18.0: Implicit dependencies on node-gyp are discouraged
```

I aim for zero-warnings because unimportant warnings make it hard to notice important warnings.

Can this warning be dealt with?

**Comment by kkoopa** [comment]
I have never used yarn, but to me it seems the warning comes from yarn and not from node-gyp. According to the [yarn documentation](https://yarnpkg.com/configuration/yarnrc#logFilters), you can set a logFilter in .yarnrc to suppress it.

**Comment by eric-g-97477** [comment]
Ok. Still seems like it is a good recommendation. You may want to reconsider the implicit dependency. Feel free to close this issue if you know otherwise or keep it open as a reminder to investigate further.

**Comment by kkoopa** [comment]
Cannot do that. They are intentionally pinned to specific versions since nothing else works across the board.

**Comment by cclauss** [comment]
Yes, but locking the version of `node-gyp` currently causes runtime failures on Python >= v3.12.
* #986 
* #985
